streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -71,6 +71,48 @@
     private static final Logger log = LoggerFactory.getLogger(StreamThread.class);
     private static final AtomicInteger STREAM_THREAD_ID_SEQUENCE = new AtomicInteger(1);
 
+    /**
+     * Stream thread states are the possible state that a stream thread can be in.
+     * A thread should only be in one state at a time
+     * The expected state transition with the following defined states is:
+     *
+     *                +-----------+
+     *                |Not Running|
+     *                +-----+-----+
+     *                      |
+     *                      v
+     *                +-----+-----+
+     *          +---->| Running   |<------------+
+     *          |     +-----+-----+             |
+     *          |           |                   |
+     *          |           v                   |
+     *          |     +-----+------------+      |
+     *          <---- |Partitions        |      |
+     *          |     |Revoked           |      |
+     *          |     +-----+------------+      |
+     *          |           |                   |
+     *          |           v                   |
+     *          |     +-----+------------+      |
+     *          |     |Partitions        |      |
+     *          |     |Assigned          |------+
+     *          |     +-----+------------+
+     *          |
+     *          |
+     *          |    +-----+----------+
+     *          +--->|Pending         |
+     *               |Shutdown        |
+     *               +-----+----------+
+     *                     |
+     *                     v
+     *               +-----+-----+
+     *               |Not Running|
```

Comment:
Stopped ? 
Is this the same state as above? If not, there should be different names. Otherwise, maybe an arrow pointing to the state about is easier to parse.